### PR TITLE
clarify the intent of the disabled field

### LINKED ISSUE #2773: publishing an unborn repository disabled for no apparent reason
If I create a new repo with an unborn branch, and with no commits, then the `Publish` button is not clickable as intended. 

However, the user can go to Repository settings and click the `Publish` button, resulting in a `Not a valid object name: 'master'` error and the nav `Publish` button still greyed out.

Reproduce:
- create new repo with no commits
- go to Repo settings, click publish
- error surfaces, but repo still published
- Publish button in nav area still unclickable

screen shot 2017-09-19 at 12 14 17

screen shot 2017-09-19 at 12 22 12

related to #2086 

tested in 1.0.0-beta3, mac

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #2773

### CODE REVIEW

File: app/src/ui/toolbar/push-pull-button.tsx [hunk 1]
```diff
@@ -32,6 +33,9 @@ interface IPushPullButtonProps {
 
   /** The current repository */
   readonly repository: Repository
+
+  /** The current state of the tip of the repository */
+  readonly tipState: TipState
```
Comment (niik) [inline app/src/ui/toolbar/push-pull-button.tsx]: I'm not sure if this documentation adds much. Could we add a note about what it's used for? I.e. to disable the push action when the current branch is unborn?

**Comment by shiftkey** [comment]
Here's what it looks like with a more relevant description - I'm not quite settled on the words, but it'd be nice if they could fit without the trimming:




The lack of contrast when this button is disabled also isn't great for the tiny description words, but that's out of scope.

**Comment by joshaber** [comment]
👉 @niik  👈

**Comment by shiftkey** [comment]
Tweaking them words to remove the ellipses (well, mostly):

**Pull request merged**